fre/cmor/cmor_mixer.py

```diff
@@ -1,408 +1,596 @@
-    var_filenames_all = os.listdir(indir)
-    print(f'(cmor_run_subtool) var_filenames_all={var_filenames_all}')
-    for var_file in var_filenames_all:
-        if var_file.endswith('.nc'):
-            var_filenames.append(var_file)
-    var_filenames.sort()
-    print(f"(cmor_run_subtool) var_filenames = {var_filenames}")
-
-
-    # name_of_set == component label, which is not relevant for CMOR/CMIP
-    name_of_set = var_filenames[0].split(".")[0]
-    print(f"(cmor_run_subtool) component label is name_of_set = {name_of_set}")
+    get_var_filenames(indir, var_filenames)
+    print(f"(cmor_run_subtool) found filenames = \n {var_filenames}")
 
+    # examine input files to obtain target date ranges
     iso_datetime_arr = []
-    for filename in var_filenames:
-        iso_datetime_arr.append(
-            filename.split(".")[1] )
-    iso_datetime_arr.sort()
-    print(f"(cmor_run_subtool) Available dates: {iso_datetime_arr}")
-    if len(iso_datetime_arr) < 1:
-        raise ValueError('ERROR: iso_datetime_arr has length 0!')
+    get_iso_datetimes(var_filenames, iso_datetime_arr)
+    print(f"(cmor_run_subtool) found iso datetimes = \n {iso_datetime_arr}")
 
-    # process each variable separately
-    for gfdl_var in gfdl_var_lst:
-        if gfdl_var_lst[gfdl_var] in proj_table_vars["variable_entry"]:
-            gfdl_to_pcmdi_var(
-                proj_table_vars, gfdl_var_lst,
-                indir, gfdl_var, iso_datetime_arr,
-                exp_config, table_config,
-                outdir, name_of_set )
-        else:
-            print(f"(cmor_run_subtool) WARNING: Skipping variable {gfdl_var} ...")
-            print( "(cmor_run_subtool)         ... it's not found in CMOR variable group")
-    print('----- END _cmor_run_subtool call -----\n\n')
+    # name_of_set == component label...
+    # which is not relevant for CMOR/CMIP... or is it?
+    name_of_set = var_filenames[0].split(".")[0]
+    print(f"(cmor_run_subtool) setting name_of_set = {name_of_set}")
+
+    # loop over entries in the json_var_list, read into var_list
+    for local_var in var_list:
+
+        # if its not in the table configurations variable_entry list, skip
+        if var_list[local_var] not in proj_table_vars["variable_entry"]:
+            print(f"(cmor_run_subtool) WARNING: skipping local_var={local_var} /"
+                  f" target_var={target_var}")
+            print( "(cmor_run_subtool)         ... target_var not found in CMOR variable group")
+            continue
+
+        # it is in there, get the name of the data inside the netcdf file.
+        target_var=var_list[local_var] # often equiv to local_var but not necessarily.
+        if local_var != target_var:
+            print(f'(cmor_run_subtool) WARNING: local_var == {local_var} '
+                  f'!= {target_var} == target_var')
+            print(f'i am expecting {local_var} to be in the filename, and i expect the variable'
+                  f' in that file to be {target_var}')
```

Comment:
I think I'm following but disagree with the message here. Shouldn't it be:

I am expecting {local_var} to be in the filename, and I expect the variable in that file to be {local_var}. The output file will be named {target_var}.